tests/alembic_migrations/versions/v_7b139906ac46_add_not_null_constraint_on_meta_vars/test_smoke.py

```diff
@@ -0,0 +1,46 @@
+"""Smoke tests:
+- Upgrade adds not null constraint
+- Downgrade drops not null constraint
+"""
+
+# -*- coding: utf-8 -*-
+import logging
+import pytest
+from alembic import command
+from sqlalchemy import inspect
+
+logger = logging.getLogger("tests")
+
+@pytest.mark.usefixtures("new_db_left")
+def test_upgrade(prepared_schema_from_migrations_left, schema_name):
+    """test migration from 3d50ec832e47 to 7b139906ac46. """
+    
+    # Set up database to version 7b139906ac46
+    engine, script = prepared_schema_from_migrations_left
+
+    # Check that views have been added
+    table = inspect(engine).get_columns("meta_vars", schema=schema_name)
+
+    for col in table:
+        if col["name"]=="cell_method" or col["name"]=="standard_name" or col["name"]=="display_name":
+            assert(col["nullable"]==False)
+
+
+@pytest.mark.usefixtures("new_db_left")
+def test_downgrade(
+    prepared_schema_from_migrations_left, alembic_config_left, schema_name
+):
+    """Test the schema migration from 7b139906ac46 to 3d50ec832e47. """
```

Comment:
And likewise a check of the downgrade data migration.

Reply:
For this one there is a call of ```command.downgrade()```, which I think means that I am testing the downgrade data migration. As far as not testing the upgrade though, I will add that in. I followed the views tests, as you noticed by my copy paste mistake, and that one did not run ```command.upgrade()``` in the ```test_upgrade``` feature, so that's why it is the way it is for me

Reply:
Yes, this means you are testing the downgrade migration -- both schema and data.

Reply:
For testing the upgrade, see my remarks above.